docs: Update enhanced meshing workflow docs. (#2859)
ansys/pyfluent: doc/source/user_guide/meshing_workflow/new_meshing_workflows.rst

.. _ref_user_guide_new_meshing_workflows:

Meshing workflow
================
You can use PyFluent to access the new, enhanced meshing workflows.

Watertight geometry meshing workflow
------------------------------------

> Use the **Watertight Geometry** workflow for water-tight CAD geometries that

Reviewer: Watertight is a word.

> do not require much in the way of cleanup or modifications.

Reviewer: "much in the way of" - aha, that come from the product doc, which we are reusing,

Reviewer: Also, this is quite useful - from the web: _This is useful for clean CADs or clean geometries that have already been prepped in another software, such as Ansys SpaceClaim._

> do not require much in the way of cleanup or modifications.

Reviewer: ```suggestion
require little cleanup.
```

> do not require much in the way of clean-up or modifications.

Reviewer 2: ```suggestion
do not require much in the way of cleanup or modifications.
```

    watertight.update_boundaries.boundary_label_type_list.set_state(["wall"])
    watertight.update_boundaries.old_boundary_label_list.set_state(["wall-inlet"])
    watertight.update_boundaries.old_boundary_label_type_list.set_state(["velocity-inlet"])
    watertight.update_boundaries()

Update regions
~~~~~~~~~~~~~~

.. code:: python

    watertight.update_regions()

Add boundary layers
~~~~~~~~~~~~~~~~~~~

.. code:: python

    watertight.add_boundary_layer.add_child_to_task()
    watertight.add_boundary_layer.insert_compound_child_task()
    watertight.task("smooth-transition_1").bl_control_name.set_state("smooth-transition_1")
    watertight.add_boundary_layer.arguments = {}
    watertight.task("smooth-transition_1")()

Generate volume mesh
~~~~~~~~~~~~~~~~~~~~

.. code:: python

    watertight.create_volume_mesh.volume_fill.set_state("poly-hexcore")
    watertight.create_volume_mesh.volume_fill_controls.hex_max_cell_length.set_state(0.3)
    watertight.create_volume_mesh()

Switch to solution mode
~~~~~~~~~~~~~~~~~~~~~~~

.. code:: python

    solver = meshing.switch_to_solver()

Fault-tolerant meshing workflow
-------------------------------

> Use the **Fault-tolerant** meshing workflow for more complicated non-water-tight CAD

Reviewer: _non-watertight_ (not hyphenated), although the sentence goes on to reiterate the point redundantly.

> geometries that may require some form of clean-up or modification (for example,

Reviewer 2: ```suggestion
geometries that might require some form of clean-up or modification (for example,
```
"May" should be used only when related to permissions, especially in regard to legal permissions.

> defects such as overlaps, intersections, holes, duplicates, etc).

Reviewer 2: ```suggestion
defects such as overlaps, intersections, holes, and duplicates).
```
Avoid Latin phrases.

> The following example shows you how to use the fault-tolerant workflow.

Reviewer 2: ```suggestion
The following example shows how to use the fault-tolerant workflow.
```

    fault_tolerant.add_boundary_layer_ftm.insert_compound_child_task()
    fault_tolerant.task("aspect-ratio_1").bl_control_name.set_state("aspect-ratio_1")
    fault_tolerant.add_boundary_layer_ftm.arguments.set_state({})
    fault_tolerant.task("aspect-ratio_1")()

Generate volume mesh
~~~~~~~~~~~~~~~~~~~~

.. code:: python

    fault_tolerant.generate_the_volume_mesh.all_region_name_list.set_state([
                "main",
                "flow_pipe",
                "outpipe3",
                "object2",
                "object1",
                "void-region-1",
                "fluid-region-1",
            ])
    fault_tolerant.generate_the_volume_mesh.all_region_size_list.set_state(["11.33375"] * 7)
    fault_tolerant.generate_the_volume_mesh.all_region_volume_fill_list.set_state(["none"] * 6 + ["tet"])
    fault_tolerant.generate_the_volume_mesh.enable_parallel.set_state(True)
    fault_tolerant.generate_the_volume_mesh()

Switch to solution mode
~~~~~~~~~~~~~~~~~~~~~~~

.. code:: python

    solver = meshing.switch_to_solver()

2D meshing workflow
-------------------
Use the **2D** meshing workflow to mesh specific two-dimensional geometries.

> The following example shows you how to use the 2D meshing workflow.

Reviewer 2: ```suggestion
The following example shows how to use the 2D meshing workflow.
```

        "no"
    )
    two_dim_mesh.generate_initial_surface_mesh.surface2_d_preferences.show_advanced_options = (
        True
    )
    two_dim_mesh.generate_initial_surface_mesh()

    two_dim_mesh.task("uniform_1").revert()
    two_dim_mesh.task("uniform_1").add_child = "yes"
    two_dim_mesh.task("uniform_1").bl_control_name = "smooth-transition_1"
    two_dim_mesh.task("uniform_1").first_layer_height = 2
    two_dim_mesh.task("uniform_1").number_of_layers = 7
    two_dim_mesh.task("uniform_1").offset_method_type = "smooth-transition"
    two_dim_mesh.task("uniform_1")()

    two_dim_mesh.generate_initial_surface_mesh.surface2_d_preferences.merge_edge_zones_based_on_labels = (
        "no"
    )
    two_dim_mesh.generate_initial_surface_mesh.surface2_d_preferences.merge_face_zones_based_on_labels = (
        "no"
    )
    two_dim_mesh.generate_initial_surface_mesh.surface2_d_preferences.show_advanced_options = (
        True
    )
    two_dim_mesh.generate_initial_surface_mesh()

Switch to solution mode
~~~~~~~~~~~~~~~~~~~~~~~

Switching to solver is not allowed in 2D Meshing mode.

Creating new meshing workflow
-----------------------------

> Use the **create_workflow** to create a custom workflow.
> The following example shows you how to use the `create_workflow`.

Reviewer 2: ```suggestion
Use the ``create_workflow()`` method to create a custom workflow.
The following example shows how to use this method.
```

Create workflow
~~~~~~~~~~~~~~~

.. code:: python

    import ansys.fluent.core as pyfluent
    from ansys.fluent.core import examples

    import_file_name = examples.download_file('mixing_elbow.pmdb', 'pyfluent/mixing_elbow')
    meshing = pyfluent.launch_fluent(
        mode="meshing", precision='double', processor_count=2
    )
    created_workflow = meshing.create_workflow()

Insert first task
~~~~~~~~~~~~~~~~~

.. code:: python

    created_workflow.insertable_tasks.import_geometry.insert()
    created_workflow.import_geometry.file_name.set_state(import_file_name)
    created_workflow.import_geometry.length_unit.set_state('in')
    created_workflow.import_geometry()

Insert next task
~~~~~~~~~~~~~~~~

.. code:: python

    created_workflow.import_geometry.insertable_tasks.add_local_sizing.insert()
    created_workflow.add_local_sizing()

Loading a saved meshing workflow
--------------------------------

> Use the **load_workflow** to create a custom workflow.
> The following example shows you how to use the `load_workflow`.

Reviewer 2: ```suggestion
Use the ``load_workflow()`` method to create a custom workflow.
The following example shows how to use this method.
```

Load workflow
~~~~~~~~~~~~~

.. code:: python

    import ansys.fluent.core as pyfluent
    from ansys.fluent.core import examples

    saved_workflow_path = examples.download_file(
        "sample_watertight_workflow.wft", "pyfluent/meshing_workflow"
    )
    meshing = pyfluent.launch_fluent(
        mode="meshing", precision='double', processor_count=2
    )
    loaded_workflow = meshing.load_workflow(file_path=saved_workflow_path)

> Inserting new task
> ------------------
> New tasks can be inserted into the meshing workflow in an object oriented manner.

Reviewer 2: ```suggestion
Insert new task
---------------
You can insert new tasks into the meshing workflow in an object-oriented manner.
```
